cmake/modules/CoverageReport.cmake

```diff
@@ -0,0 +1,73 @@
+# if coverage reports are not enabled, skip all of this
+if(NOT LLVM_BUILD_INSTRUMENTED_COVERAGE)
+  return()
+endif()
+
+file(TO_NATIVE_PATH
+     "${LLVM_SOURCE_DIR}/utils/prepare-code-coverage-artifact.py"
+     PREPARE_CODE_COV_ARTIFACT)
+
+# llvm-cov and llvm-profdata need to match the host compiler. They can either be
+# explicitly provided by the user, or we will look them up based on the install
+# location of the C++ compiler.
+get_filename_component(COMPILER_DIRECTORY ${CMAKE_CXX_COMPILER} DIRECTORY)
+find_program(LLVM_COV "llvm-cov" ${COMPILER_DIRECTORY} NO_DEFAULT_PATH)
+find_program(LLVM_PROFDATA "llvm-profdata" ${COMPILER_DIRECTORY} NO_DEFAULT_PATH)
+
+if(NOT LLVM_COV OR NOT LLVM_PROFDATA)
+  message(WARNING "Could not find code coverage tools, skipping generating targets. You may explicitly specify LLVM_COV and LLVM_PROFDATA to work around this warning.")
+  return()
+endif()
+
+set(LLVM_CODE_COVERAGE_TARGETS "" CACHE STRING "Targets to run code coverage on (defaults to all exported targets if empty)")
+mark_as_advanced(LLVM_CODE_COVERAGE_TARGETS)
+
+# HLSL Change Begin - This is probably worth upstreaming...
+set(LLVM_CODE_COVERAGE_TEST_TARGETS "" CACHE STRING "Targets to run to generate coverage profiles.")
+mark_as_advanced(LLVM_CODE_COVERAGE_TEST_TARGETS)
```

Comment:
In the past, I remember using a bootstrap process to generate profiles.  So the target would be cc1/cc1plus itself.  In here, isn't it just `dxc`?

Reply:
We don't (yet) have any build targets inside the CMake build that run `dxc` or its tests. For code coverage the goal is really to get the coverage numbers from running the tests, which at the moment run through external testing harnesses, but will soon run under CMake and Lit (#4762).

Reply:
In my local runs I set `LLVM_CODE_COVERAGE_TARGETS` to `dxc;dxcompiler`, which constrains the coverage report to just DXC and the compiler dynamic library, but we instrument all the binaries so that the test binaries report coverage for the code they load from `dxcompiler`.

Reply:
Wait, is `LLVM_CODE_COVERAGE_TARGETS` meant to be targets like `dxc` or something along the lines of `check-dxc`? Your explanation here sounds like the former, but the logic it replaces is more like the latter.